### ISSUE TITLE
"proc-macro derive panicked" when deriving WorldQuery.

### ISSUE THREAD

**Issue by linkpy** [bevyengine/bevy#4533]
## Bevy version

0.7.0

## Operating system & version

Pop_OS! 21.10

## What you did

```rs
#[derive(Component)]
struct Comp1;

#[derive(Component)]
struct Comp2;

#[derive(Component)]
struct Comp3;

#[derive(WorldQuery)]
struct WithoutComps {
  _1: Without<Comp1>,
  _2: Without<Comp2>,
  _3: Without<Comp3>
}
```

## What you expected to happen

The project to compile.

## What actually happened

```
$ cargo run
   Compiling somegame v0.1.0 (/home/pancake/Documents/dev/rust/somegame)
error: proc-macro derive panicked
  --> src/main.rs:29:10
   |
29 | #[derive(WorldQuery)]
   |          ^^^^^^^^^^
   |
   = help: message: called `Option::unwrap()` on a `None` value
```

**Comment by bjorn3** [comment]
I'm getting

```rust
error: proc-macro derive panicked
  --> src/main.rs:16:10
   |
16 | #[derive(WorldQuery)]
   |          ^^^^^^^^^^
   |
   = help: message: Expected a struct with a lifetime
```

instead when using `bevy_ecs` directly.

**Comment by CGMossa** [comment]
With the help of `@Gibonus` on discord, the fix for this is: 

```rust
#[derive(WorldQuery)]
struct WithoutComps<'w> {
  _1: Without<Comp1>,
  _2: Without<Comp2>,
  _3: Without<Comp3>,
  #[world_query(ignore)]
  _fix: Option<&'w ()>
}
```

**Comment by BoxyUwU** [comment]
This has been fixed by #3001
